common/processes.hpp

```diff
@@ -0,0 +1,52 @@
+// Copyright 2023 Northern.tech AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#include <string>
+#include <vector>
+#include <common/error.hpp>
+#include <common/expected.hpp>
+
+#ifndef MENDER_COMMON_PROCESSES_HPP
+#define MENDER_COMMON_PROCESSES_HPP
```

Comment:
Yeah, functionally it doesn't really matter. Maybe it will be *slightly* more work for the compiler if outside, but it's not like that makes a difference. To me it just felt natural that the guards would surround everything, but either is fine with me.

Reply:
Yes, I agree with Kristian here @vpodzime